File: sol/solution/0001-0100/0024/Solution.php
/**
 * Definition for a singly-linked list.
 * class ListNode {
 *     public $val = 0;
 *     public $next = null;
 *     function __construct($val = 0, $next = null) {
 *         $this->val = $val;
 *         $this->next = $next;
 *     }
 * }
 */
class Solution {
    /**
     * @param ListNode $head
     * @return ListNode
     */
    function swapPairs($head) {
        if (!$head || !$head->next) return $head;
        $dummy = new ListNode(0, $head);
        $prev = $dummy;
        while ($prev->next && $prev->next->next) {
            $first = $prev->next;
            $second = $first->next;
            $first->next = $second->next;
            $second->next = $first;
            $prev->next = $second;
            $prev = $first;
        }
        return $dummy->next;
    }
}

File: sol/solution/0001-0100/0024/Solution.scala
/**
 * Definition for singly-linked list.
 * class ListNode(_x: Int = 0, _next: ListNode = null) {
 *   var next: ListNode = _next
 *   var x: Int = _x
 * }
 */
object Solution {
    def swapPairs(head: ListNode): ListNode = {
        if (head == null || head.next == null) return head
        val dummy = new ListNode(0, head)
        var prev: ListNode = dummy
        while (prev.next != null && prev.next.next != null) {
            val first = prev.next
            val second = first.next
            first.next = second.next
            second.next = first
            prev.next = second
            prev = first
        }
        dummy.next
    }
}